test/common.js

```diff
@@ -354,3 +354,61 @@ exports.fileExists = function(pathname) {
     return false;
   }
 };
+
+
+
+(function() {
+  var asyncTestsEnabled = false;
+  var asyncTestLastCheck;
+  var asyncTestQueue = [];
+  var asyncTestHandle;
+  var currentTest = null;
+
+  function fail(error) {
+    var stack = currentTest
+          ? error.stack + "\nFrom previous event:\n" + currentTest.stack
+          : error.stack;
+
+    process.stderr.write(stack);
+    process.exit(2);
+  }
+
+  function nextAsyncTest() {
+    var called = false;
+    function done(err) {
+      if (called) return fail(new Error("done called twice"));
+      called = true;
+      asyncTestLastCheck = Date.now();
+      if (arguments.length > 0) return fail(err);
+      setTimeout(nextAsyncTest, 10);
+    }
+
+    if (asyncTestQueue.length) {
+      var test = asyncTestQueue.shift();
+      currentTest = test;
+      test.action(done);
+    } else {
+      clearInterval(asyncTestHandle);
+      process.exit(0);
```

Comment:
I'm not a fan of calling process.exit() in tests.  Historically, quite a few bugs have gone unnoticed because a test called process.exit().